### PR TITLE
[flang][Driver] Support -Xlinker in flang

### PR DESCRIPTION
**Pull request by tarunprabhu**
Partially addresses: https://github.com/llvm/llvm-project/issues/89888

### CODE REVIEW

File: flang/test/Driver/xlinker.f90 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+! RUN: %flang -### -o /dev/null %s -Xlinker -rpath=/not/a/real/path 2>&1 | FileCheck %s
+
+! CHECK: "-fc1"
+! CHECK-NEXT: "-rpath=/not/a/real/path"
```
Comment (tblah) [inline flang/test/Driver/xlinker.f90:4]: nit: maybe I'm being overly pedantic here, but I think the meaning of the test would be clearer if you check directly that this is an argument to the linker, not assuming that the next line after the frontend driver is the linker. There are some regular expressions for names of the linker in `flang/test/Driver/linker-flags.f90`.

**Review by tblah — APPROVED** [review]

File: flang/test/Driver/xlinker.f90 [hunk 2]
```diff
@@ -0,0 +1,8 @@
+! RUN: %flang -### -o /dev/null %s -Xlinker -rpath=/not/a/real/path 2>&1 | FileCheck --check-prefix=SINGLE %s
+! RUN: %flang -### -o /dev/null %s -Xlinker -rpath -Xlinker /not/a/real/path 2>&1 | FileCheck --check-prefix=MULTIPLE %s
+
+
+! SINGLE: "-rpath=/not/a/real/path"
+! MULTIPLE: "-rpath" "/not/a/real/path"
```
Comment (tblah) [inline flang/test/Driver/xlinker.f90]: nit: The problem with this is it does not check that the -Xlinker flag is actually passed to the linker

**Review by tblah — APPROVED** [review]
Thanks for the updates. LGTM

File: flang/test/Driver/xlinker.f90 (hunk 1 above)
Comment (tarunprabhu) [reply flang/test/Driver/xlinker.f90]: Thanks. I have updated the tests quite extensively. There are now checks for various platforms and multiple occurrences of `-Xlinker`. How does this look?

**Comment by tarunprabhu** [comment]
> Thanks for the updates. LGTM

Thanks for the reviews, Tom :-)

File: flang/test/Driver/xlinker.f90 (hunk 2 above)
Comment (tarunprabhu) [reply flang/test/Driver/xlinker.f90]: Yes. I realized that soon immediately after I pushed up the changes (sigh). Thanks for taking multiple looks.

**Pull request merged**